epic-document-api/src/epic_document_api/services/object_storage_service.py

```diff
@@ -17,75 +20,118 @@ class ObjectStorageService:
     def __init__(self):
         """Initialize the service."""
         # initialize s3 config from environment variables
-        self.s3_access_key_id = current_app.config.get('S3_ACCESS_KEY_ID')
-        self.s3_secret_access_key = current_app.config.get('S3_SECRET_ACCESS_KEY')
-        self.s3_host = current_app.config.get('S3_HOST')
-        self.s3_bucket = current_app.config.get('S3_BUCKET')
-        self.s3_region = current_app.config.get('S3_REGION')
-        self.s3_service = current_app.config.get('S3_SERVICE')
-
-    def get_url(self, filename: string, folder: string = '') -> string:
+        self.s3_access_key_id = current_app.config.get("S3_ACCESS_KEY_ID")
+        self.s3_secret_access_key = current_app.config.get("S3_SECRET_ACCESS_KEY")
+        self.s3_host = current_app.config.get("S3_HOST")
+        self.s3_bucket = current_app.config.get("S3_BUCKET")
+        self.s3_region = current_app.config.get("S3_REGION")
+        self.s3_service = current_app.config.get("S3_SERVICE")
+        self.s3_client = Boto3ClientFactory(
+            endpoint_url=f"https://{current_app.config.get('S3_HOST')}"
+        ).get_client("s3")
+
+    def get_url(self, filename: string, folder: string = "") -> string:
         """Get the object url."""
         if not all([self.s3_host, self.s3_bucket, filename]):
-            return ''
+            return ""
 
         if folder:
-            folder = folder.strip('/')
-            folder = f'/{folder}/'
+            folder = folder.strip("/")
+            folder = f"/{folder}/"
         else:
-            folder = '/'
+            folder = "/"
 
-        return f'https://{self.s3_host}/{self.s3_bucket}{folder}{filename}'
+        return f"https://{self.s3_host}/{self.s3_bucket}{folder}{filename}"
 
     def delete_s3_object(self, file: Dict) -> Dict:
         """Delete the S3 object."""
         self._check_s3_configuration()
-        s3_source_uri = file.get('filepath')
+        s3_source_uri = file.get("filepath")
 
         auth = self._create_aws_auth()
-        response = self._make_s3_auth_headers_request(s3_source_uri, '', auth)
+        response = self._make_s3_auth_headers_request(s3_source_uri, "", auth)
         auth_headers = response.request.headers
         self._make_s3_delete_request(s3_source_uri, auth, auth_headers)
-        document = Document.get_by_path(s3_source_uri)
+        document = DocumentModel.get_by_path(s3_source_uri)
         if document:
             document.delete()
         return file
 
+    def generate_presigned_urls(self, payload: list[dict]):
```

Comment:
any chance we can breakout the logic for PUT and DELETE to separate functions and just call them from here.
This function may be doing a lot of things. I'm referring to the Single Responsibility Principle here.